dask_cuml/linear_model/linear_regression.py

```diff
@@ -0,0 +1,560 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf
+import dask_cudf
+import numpy as np
+
+from dask import delayed
+from dask_cuml.core import new_ipc_thread, parse_host_port
+from dask_cuml.core import device_of_devicendarray, build_host_dict
+from dask.distributed import wait, default_client
+from math import ceil
+from numba import cuda
+from toolz import first
+from tornado import gen
+
+
+class LinearRegression(object):
+    """
+    Model-Parallel Multi-GPU Linear Regression Model. Single Process Multi GPU
+    supported currently
+    """
+    def __init__(self, fit_intercept=True, normalize=False):
+
+        """
+        Initializes the linear regression class.
+
+        Parameters
+        ----------
+        fit_intercept: boolean. For more information, see `scikitlearn's OLS
+        <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+        normalize: boolean. For more information, see `scikitlearn's OLS
+        <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+
+        """
+        self.coef_ = None
+        self.intercept_ = None
+        self.fit_intercept = fit_intercept
+        self.normalize = normalize
+        self._model_fit = False
+        self._consec_call = 0
+
+    def _build_params_map(self):
+        return {"fit_intercept": self.fit_intercept,
+                "normalize": self.normalize}
+
+    def fit(self, X, y):
+        """
+        Fits a multi-gpu linear regression model such that each the resulting
```

Comment:
Should this read "such that each 'of' the..."?

Reply:
I can't English :)